### PR TITLE
Add New Operations and UI schema

### PR DESCRIPTION
**Pull request by niruhan**
## Purpose
Add the following operations

1. convert-lead
2. find-duplicate
3. find-duplicates-by-ids
4. merge
5. get-server-timestamp
6. get-deleted
7. get-updated

### CODE REVIEW

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        if (synLog.isTraceOrDebugEnabled()) {
+            synLog.traceOrDebug("End : Salesforce Find Duplicates mediator");
+
+            if (synLog.isTraceTraceEnabled()) {
+                synLog.traceTrace("Message : " + synCtx.getEnvelope());
+            }
+        }
+    }
+}
```
Comment (dinuish94) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:51]: Let's add a new line at the end of the file in all the files

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (dinuish94) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:2]: Update the license header in all the files

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SalesforceUtil.java [hunk 3]
```diff
@@ -39,9 +39,12 @@
     public static final String SALESFORCE_CRUD_ALLOWFIELDTRUNCATE = "allowFieldTruncate";
     public static final String SALESFORCE_EXTERNALID = "externalId";
     public static final String SALESFORCE_SOBJECTS = "sobjects";
+    public static final String SALESFORCE_REQUEST = "mergerequests";
```
Comment (abeykoon) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SalesforceUtil.java:42]: should we rename this as SALESFORCE_MERGE_REQUEST?

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * under the License.
+ */
+
+package org.wso2.carbon.connector.salesforce;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseLog;
+import org.wso2.carbon.connector.core.AbstractConnector;
+
+public class SetupConvertLead extends AbstractConnector {
```
Comment (abeykoon) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:25]: Shall we add class level comment here?

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        if (synLog.isTraceOrDebugEnabled()) {
+            synLog.traceOrDebug("Start : Salesforce Find Duplicates mediator");
+
+            if (synLog.isTraceTraceEnabled()) {
+                synLog.traceTrace("Message : " + synCtx.getEnvelope());
+            }
+        }
+
+        SalesforceUtil salesforceUtil = SalesforceUtil.getSalesforceUtil();
```
Comment (abeykoon) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:39]: Should util methods be static? Why we need an instance?

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        salesforceUtil.addSobjects("convertLead", SalesforceUtil.SALESFORCE_LEAD_CONVERT, synCtx, synLog,
+                null);
+
+        if (synLog.isTraceOrDebugEnabled()) {
```
Comment (abeykoon) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:43]: Is this two nested related to logs are common? What is the difference between the two? I am thinking of simplifying and moving to a common method.

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import org.wso2.carbon.connector.core.AbstractConnector;
+
+public class SetupConvertLead extends AbstractConnector {
+
+    public void connect(MessageContext synCtx) {
+
+        SynapseLog synLog = getLog(synCtx);
+
+        if (synLog.isTraceOrDebugEnabled()) {
+            synLog.traceOrDebug("Start : Salesforce Find Duplicates mediator");
```
Comment (abeykoon) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:32]: What is this `Find Duplicates mediator`?  This is not finding duplicates, is it?

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupFindDuplicatesByIds.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+ * under the License.
+ */
+
+package org.wso2.carbon.connector.salesforce;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseLog;
+import org.wso2.carbon.connector.core.AbstractConnector;
+
+public class SetupFindDuplicatesByIds extends AbstractConnector {
```
Comment (abeykoon) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupFindDuplicatesByIds.java:25]: Shall we add class level comments to all the classes as applicable?

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java (hunk 2 above)
Comment (abeykoon) [inline connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:2]: For the new classes, copyright should be updated to 2020.

Comment (abeykoon) [reply connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:2]: The Copyright format could have been also updated. Please check.

File: connector/src/main/resources/salesforce/convertLead.xml [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+                            <urn:sessionId>$1</urn:sessionId>
+                        </urn:SessionHeader>
+                    </soapenv:Header>
+                    <soapenv:Body>
+                        <urn:convertLead/>
+                    </soapenv:Body>
+                </soapenv:Envelope>
+            </format>
+            <args>
+                <arg expression="get-property('operation','salesforce.sessionId')"/>
```
Comment (abeykoon) [inline connector/src/main/resources/salesforce/convertLead.xml:41]: What is the need to use `operation` context? Cannot we use `default` (synapse) context?

Comment (abeykoon) [reply connector/src/main/resources/salesforce/convertLead.xml:41]: Also instead of `get-property`, better  use `$ctx:` the new syntax if we can use synapse context.

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java (hunk 6 above)
Comment (niruhan) [reply connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:43]: To indicate start and end of the class mediator operation. One for start and another for end

File: connector/src/main/resources/salesforce/convertLead.xml (hunk 9 above)
Comment (niruhan) [reply connector/src/main/resources/salesforce/convertLead.xml:41]: In the sf soap init they have used operation context

File: connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java (hunk 6 above)
Comment (abeykoon) [reply connector/src/main/java/org/wso2/carbon/connector/salesforce/SetupConvertLead.java:46]: +1. We will add it to connector improvement doc.

File: connector/src/main/resources/salesforce/convertLead.xml (hunk 9 above)
Comment (abeykoon) [reply connector/src/main/resources/salesforce/convertLead.xml:41]: We'll keep it for the moment. However, we'll evaluate
